### PR TITLE
Add shortcuts.addShortcuts to add shortcuts

### PR DESCRIPTION
**Pull request by iwehrman**
1. Add a new `shortcuts.addShortcuts` action to add shortcuts in bulk.
2. Use one call to `shortcuts.addShortcuts` in `tools.beforeStartup`. Previously, we were doing separate transfers for each call to `shortcuts.addShortcut` (singular) which was both bad for performance (N redundant policy calls in the adapter instead of one) and a concurrency-safety problem waiting to happen. 

Addresses: concerns.

### CODE REVIEW

File: src/js/actions/tools.js [hunk 1]
```diff
@@ -695,37 +695,43 @@ define(function (require, exports) {
...
-            return promises;
+            return specs;
         }.bind(this), []);
     
         _vectorMaskHandler = function () {
             var toolStore = this.flux.store("tool"),
                 vectorMode = toolStore.getVectorMode() || false;
             
-            return this.transfer(changeVectorMaskMode, !vectorMode);
+            this.flux.actions.tools.changeVectorMaskMode(!vectorMode);
```
Comment (iwehrman) [inline src/js/actions/tools.js]: This is an unrelated fix. It's not correct to transfer from an event handler like this because the locks are no longer being held. Instead, a new action should be enqueued.

**Comment by shaoshing** [comment]
LGTM! :+1:
